File: Prefix Sum/Equilibrium Point.py
# User function Template for python3

class Solution:
    #Function to find equilibrium point in the array.
    def findEquilibrium(self, arr):
        # code here
        n=len(arr)
        totalSum=sum(arr)
        leftSum=0
        
        for i in range(n):
            totalSum-=arr[i]
            if leftSum==totalSum:
                return i
            leftSum+=arr[i]
        return -1

File: Prefix Sum/C++/Equilibrium Point.cpp
class Solution {
  public:
    int findEquilibrium(vector<int> &arr) {
        // code here
        int prefSum = 0;
        int total = accumulate(arr.begin(), arr.end(), 0);
        
        for (int i = 0; i < arr.size(); i++) {
            int sufSum = total - prefSum - arr[i];
            if (prefSum == sufSum)
                return i;
            prefSum += arr[i];
        }
        return -1;
    }
};
